Automattic/wp-calypso
# Domains Management Revamp: Add domain contact info edit page

Closes #96818

## Proposed Changes

* Adds the sub-page with the domain contact info edit form. The page header is slightly different from the design as explained [here](https://github.com/Automattic/wp-calypso/pull/97368).
* Visual improvement to the `SubpageWrapper` component.
* Refactor `EditContactInfoPage` by extracting some of the logic to a new component - `EditContactInfoPageContent`.

## Why are these changes being made?


This is part of the Domain Management Revamp project: pfuQfP-13x-p2

## Testing Instructions



* Checkout the branch.
* Go to http://calypso.localhost:3000/domains/manage?flags=calypso/all-domain-management.
* Click on a domain where you can edit the contact info.
* Go to the "Contact Information" section and click the "Edit" button.
* You will be redirected to http://calypso.localhost:3000/domains/manage/all/contact-info/edit/[site]/[domain]
* Compare the page with the design - ePfz53j71KKDLWtuSm567A-fi-4949_49077. Note that the header is different, as explained previously.
* Check that the page looks good on mobile.
* Submit the form and make sure it works.
* Go to the old view - http://calypso.localhost:3000/domains/manage/[site]/edit-contact-info/[domain].
* Make sure it looks and works as before.

![image](https://github.com/user-attachments/assets/fa755002-5652-4fbb-ae19-d22a734271eb)

New view:
![image](https://github.com/user-attachments/assets/1771e33b-d2b1-40de-8341-5507632b1e92)

Old view:
![image](https://github.com/user-attachments/assets/b024e8dd-bc2e-4845-9f53-ca5231ce50cf)


## Pre-merge Checklist
